Maskbook/Scene/App/RedPacket/ViewModel/LuckyDropViewModel.swift

```diff
@@ -0,0 +1,728 @@
+//
+//  LuckyDropViewModel.swift
+//  Maskbook
+//
+//  Created by Hugo L on 2022/3/29.
+//  Copyright © 2022 dimension. All rights reserved.
+//
+
+import BigInt
+import Combine
+import CoreDataStack
+import Foundation
+import SwiftUI
+import SwiftyJSON
+import web3swift
+import PromiseKit
+import CombineExt
+
+class LuckyDropViewModel: NSObject, ObservableObject {
+    // MARK: - Public property
+    @Published var quantityStr = ""
+    @Published var amountStr = ""
+    @Published var message = L10n.Plugins.Luckydrop.Buttons.bestWishes
+    @Published var mode = RedPacket.RedPacketType.average {
+        didSet {
+            quantityStr = ""
+            amountStr = ""
+        }
+    }
+    @Published var token: Token? {
+        didSet {
+            amountStr = ""
+        }
+    }
+    @Published var gasFeeItem: GasFeeCellItem?
+    @Published var buttonType: ConfirmButtonType = .send
+    @Published var allowances = [String: BigUInt]()
+    @Published var selection = LuckDropKind.token
+    
+    // TODO: It needs to init `profileNickName` if create red packet from Social Platform.
+    let profileNickName: String? = nil
+    let walletBottomViewModel = WalletBottomWidgetViewModel()
+    
+    var gasFeeViewModel = GasFeeViewModel()
+    
+    var tokenStr: String {
+        token?.symbol ?? ""
+    }
+    
+    var maxButtonEnable: Bool {
+        guard mode == .average else {
+            return true
+        }
+        
+        guard let quantity = Int(quantityStr), let amount = token?.quantity?.doubleValue else {
+            return false
+        }
+        return quantity > 0 && amount > 0
+    }
+    
+    var gasFeeInfo: String {
+        guard let gasFeeItem = gasFeeItem else {
+            return ""
+        }
+        
+        guard let gasLimt = gasFeeViewModel.localGasFeeModel?.gasLimit else {
+            return ""
+        }
+        
+        guard let tokenPrice = token?.price as? Double else {
+            return ""
+        }
+        
+        let symbol = maskUserDefaults.currency.symbol
+        let gwei = gasFeeItem.gWei
+        let gasPriceDoubleValue = Double(EthUtil.getGasFeeFiat(gwei: gwei, gasLimit: gasLimt, price: tokenPrice)) ?? 0
+        var gasPrice: String
+        if gasPriceDoubleValue < 0.01 {
+            gasPrice = "< \(symbol)0.01"
+        } else {
+            gasPrice = "~\(symbol)\(EthUtil.getGasFeeFiat(gwei: gwei, gasLimit: gasLimt, price: tokenPrice))"
+        }
+        let time = gasFeeItem.shortCostTime
+        
+        return "\(gasPrice) (\(time))"
+    }
+    
+    var gasLimit: BigUInt {
+        guard let gasLimitStr = gasFeeViewModel.gasFeePublisher.value?.gasLimit else {
+            return BigUInt(21_000)
+        }
+        return BigUInt(gasLimitStr) ?? BigUInt(21_000)
```

Comment:
May be use a static variable, which named as `defaultGasLimit`, to get the default gasLimit.